### PR TITLE
Basic redux (cleaner branch)

### PR DESCRIPTION
**Pull request by Spetastian**
This adds a basic Redux flow. For now only the messages and handling change of username is done via Redux. This is related to https://github.com/cryptag/leapchat/issues/91

Also, I added a basic `.editorconfig` and `.eslintrc` config for consistent intendation using 2 spaces. I did not add a lot of stuff in the `.eslintrc`, it shuld evolve over time while we descide on code standards and change the code accordingly.

### CODE REVIEW

**Comment by elimisteve** [comment]
@Spetastian I get:

```
build-05bd8599d1.min.js:40 Uncaught ReferenceError: compose is not defined
```

File: src/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React, { Component } from 'react';
 import ReactDOM from 'react-dom';
-import { Provider } from 'react-redux';
+import { Provider, compose } from 'react-redux';
```
Comment (elimisteve) [inline src/index.js]: @Spetastian This fixed part of it, but now I get:

```
Uncaught TypeError: h is not a function
    at Object.r.529../components/App (build-7ee3a451c8.min.js:40)
    at o (build-7ee3a451c8.min.js:16)
    at e (build-7ee3a451c8.min.js:16)
    at build-7ee3a451c8.min.js:16
r.529../components/App @ build-7ee3a451c8.min.js:40
o @ build-7ee3a451c8.min.js:16
e @ build-7ee3a451c8.min.js:16
(anonymous) @ build-7ee3a451c8.min.js:16
```

The code that `compose` transpiles to calls `h()` but it's not defined for some reason? Hmm...

Comment (Spetastian) [reply src/index.js]: Oh sorry, it is when connecting to Redux dev tools:
`const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;` 

I copied that from another project. I do no use compose yet and that is why i removed it. I missed out on that it is used on that line.

Comment (Spetastian) [reply src/index.js]: @elimisteve I get exactly the same. Even though i have changed the setup for Redux dev tools according to how it should be done after some updates. They have changed it a bit. 

This was not a problem using webpack, actually if you use webpack with hot module replacement , then Redux dev tools can hot reload Reducers. I guess we can skip using Redux dev tools until we have a good setup with webpack?

**Comment by Spetastian** [comment]
@elimisteve There is actually still a problem with `compose` though. 

I get exactly the same. Even though i have changed the setup for Redux dev tools according to how it should be done after some updates. They have changed it a bit.

This was not a problem using webpack, actually if you use webpack with hot module replacement , then Redux dev tools can hot reload Reducers. I guess we can skip using Redux dev tools until we have a good setup with webpack?

Maybe we should just skip using Redux dev tools for now?

**Comment by Spetastian** [comment]
@elimisteve So we are not currently using the compose method and Redux devtools integration. Also, I added eslint rules and editorconfig to make sure to not mix tabs with spaces and keep it at 2 spaces.

**Comment by elimisteve** [comment]
Just ran `npm install` and got this:

```
$ npm install
LeapChat@0.1.0 /home/user/gocode/src/github.com/cryptag/leapchat
├─┬ react-redux@5.0.5 
│ ├── create-react-class@15.6.0 
│ ├── hoist-non-react-statics@1.2.0 
│ ├── lodash-es@4.17.4 
│ └── prop-types@15.5.10 
├─┬ redux@3.7.2 
│ └── symbol-observable@1.0.4 
├─┬ redux-devtools@3.4.0 
│ └── redux-devtools-instrument@1.8.2 
├─┬ redux-devtools-dock-monitor@1.1.2 
│ ├── parse-key@0.2.1 
│ ├─┬ react-dock@0.2.4 
│ │ └── lodash.debounce@3.1.1 
│ └── react-pure-render@1.0.2 
└─┬ redux-devtools-log-monitor@1.3.0 
  ├── lodash.debounce@4.0.8 
  ├─┬ react-json-tree@0.10.9 
  │ └─┬ react-base16-styling@0.5.3 
  │   ├── lodash.curry@4.1.1 
  │   ├── lodash.flow@3.5.0 
  │   └── pure-color@1.3.0 
  └─┬ redux-devtools-themes@1.0.0 
    └── base16@1.0.0 

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN browserify-shim@2.0.10 requires a peer of browserify@>= 2.3.0 < 4 but none was installed.
```

Notice that last line.  Think that's fine? @Spetastian

**Comment by Spetastian** [comment]
@elimisteve Regarding the warning about fsevent. That should be fine. Read this: https://stackoverflow.com/questions/31228862/npm-warn-optional-dep-failed-continuing-fsevents0-3-6

At work we have this warning in all of our build logs at work for every service.

File: src/components/chat/ChatContainer.js [hunk 2]
```diff
@@ -5,25 +5,25 @@ import MessageForm from './MessageForm';
...
+          onAlertDismiss={onAlertDismiss} />
 
         <MessageBox
           messages={messages}
           username={username} />
 
         <MessageForm
-          onSendMessage={onSendMessage} />
+          onSendMessage={onSendMessage}
+          hasFocus={this.props.messageInputFocus} />
```
Comment (elimisteve) [inline src/components/chat/ChatContainer.js]: @Spetastian How about `shouldHaveFocus`?  I think that's what we're using this for, correct?

File: src/index.js [hunk 3]
```diff
@@ -1,6 +1,30 @@
...
+import { createStore } from 'redux';
+import rootReducer from './reducers';
 import App from './components/App';
+import ReduxDevTools from './components/_dev/ReduxDevTools';
+
+/*
+const enhancer = compose(
+  ReduxDevTools.instrument()
+);
+*/
```
Comment (elimisteve) [inline src/index.js:13]: @Spetastian What will this do?  (If you think it's a good reminder to have this here, we don't need to delete it.)

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            PGPORT: 5432
+            PGDATABASE: leapchat
+            PGSCHEMA: public
+            DB_ANON_ROLE: postgres
+        depends_on:
+            - postgres
+    adminer:
+        image: adminer
+        ports:
+            - 8081:8080
```
Comment (elimisteve) [inline docker-compose.yml:31]: @Spetastian Can you add instructions to `README.md` for using this Docker Compose to run LeapChat using this file?

File: src/index.js (hunk 3 above)
Comment (Spetastian) [reply src/index.js:13]: It is a reminder. Once the webpack pipeline is at place this will work fine.

File: docker-compose.yml (hunk 4 above)
Comment (Spetastian) [reply docker-compose.yml:31]: Yes, of course.

File: src/components/chat/ChatContainer.js (hunk 2 above)
Comment (Spetastian) [reply src/components/chat/ChatContainer.js]: Sure, that is a more descriptive name of what the property is doing. More declarative that way.

**Comment by elimisteve** [comment]
:tada: Great work!  Thank you much, @Spetastian :-D.

**Pull request merged**